src/storage/hummock_trace/README.md

```diff
@@ -0,0 +1,77 @@
+# Hummock trace and replay
+
+## Tracing
+
+
+### Risedev
+
+Put this env variable in `risedev-components.user.env`
+```toml
+ENABLE_HM_TRACE=true
+```
+It makes `risingdev` put flag `hm_trace` in env variables.
+
+You can also config written log path
+```toml
+HM_TRACE_PATH=".trace/hummock.ht"
+```
+
+Then running any risedev commands traces storage operations to the log file.
+
+### CLI
+If you wish to manually run `cargo` rather than `risedev`, set the env variable to enable tracing.
+```
+RUSTFLAGS="--cfg hm_trace --cfg tokio_unstable"
+```
+We must manually enable `tokio_unstable` because extra flag sources are mutually exclusive. If we provide this variable, cargo will not evaluate `build.rustflags` in `.cargo/config.toml`
+
+For example, to start a traced playground
+
+```
+RUSTFLAGS="--cfg hm_trace --cfg tokio_unstable" cargo run --bin risingwave playground
+```
+
+### Development
+It's recommended to add `--cfg hm_trace` flag to `.cargo/config.toml` for development.
+Example:
+```toml
+[target.'cfg(all())']
+rustflags = [
+  "--cfg",
+  "hm_trace"
+]
+```
+
+## Replay
+
+### Config
+
+Replaying requires the complete object storage from tracing. Please configure you object storage in your config file and make sure data remain in object storage.
+
+Example:
+```toml
+[storage]
+data_directory = "hummock_001"
+local_object_store = "minio://hummockadmin:hummockadmin@host/hummock001"
+```
+
+### Sstable Offset hacking
+
+In `src/storage/hummock/compactor/mod.rs` function `open_builder`, set
+```rust
+let table_id = self.sstable_id_manager.get_new_sst_id().await? + 2147483647; // add a large enough offset
```

Comment:
People may forget to make this modification and accidentally corrupt existing data so requiring modifying the codebase before running the replay tool is not a good idea. Let's workaround it in other ways.

Reply:
> 
Can be solved by adding `sst_offset` to `MockHummockMetaClient`. It's ok because we only use it for tests.

https://github.com/odysa/risingwave/blob/b2973097edce967f34d89645bbb742c219e017ca/src/meta/src/hummock/mock_hummock_meta_client.rs#L122-L135